### ISSUE TITLE
Dimensions check in BinaryCrossEntropy

### ISSUE THREAD

**Issue by AakashKumarNain** [tensorflow/tensorflow#30040]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary):Binary
- TensorFlow version (use command below):2.0.0-beta1
- Python version:3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
Suppose we build a model for binary classification problem and we want to use `BinaryCrossEntropy` loss provided in `tf.keras.losses`. Here is an example:
```python
import numpy as np
import tensorflow as tf

y_true = np.array([[1.], [1.], [1.], [0.], [1.], [0.], [0.], [1.], [1.], [0.]]).astype(np.float32)
y_pred = np.array([[0.], [0.], [0.], [1.], [1.], [0.], [0.], [1.], [0.], [1.]]).astype(np.float32)

print(y_true.shape, y_pred.shape) # prints (10, 1) (10, 1)

# loss function
bce = tf.keras.losses.BinaryCrossentropy()

# Case1:
print(bce(y_true, y_pred).numpy())  # prints 9.23662 correctly

# Case2:
print(bce(np.squeeze(y_true), y_pred).numpy()) # prints 8.006299
```

**Describe the expected behavior**
When the `dimensions` of `y_true` and `y_pred` are different, in that case the loss function should raise an error for `dimension` mismatch or the model will fail **silently** and no one would be ab;e to debug it until unless they are aware of this behavior 

**Code to reproduce the issue**
Check above

**Comment by yongtang** [comment]
Added a PR #30053 for the shape check.
